lib/puppet/util/run_mode.rb

```diff
@@ -55,11 +55,19 @@ def which_dir( system, user )
 
     class UnixRunMode < RunMode
       def conf_dir
-        which_dir("/etc/puppet", "~/.puppet")
+        which_dir("/etc/puppetlabs/agent", "~/.puppet")
       end
 
       def var_dir
-        which_dir("/var/lib/puppet", "~/.puppet/var")
+        which_dir("/opt/puppetlabs/agent/cache", "~/.puppet/var")
+      end
+
+      def run_dir
+        which_dir("/var/run/puppetlabs", "~/.puppet/var/run")
+      end
+
+      def log_dir
+        which_dir("/var/log/puppetlabs/agent", "~/.puppet/var/log/")
```

Comment:
The `logdir` and `rundir` fall under `vardir` if I don't specify `vardir` (when starting the master as user nwolfe via `puppet master --no-daemonize ...`), but only because they happen to be hard-coded under `~/.puppet/var`.

It'd be nice if we could interpolate these to be under $vardir.  This would only apply when we're non-root since the paths *don't* follow this structure when run as root.